src/services/messageservice/message.service.ts

```diff
@@ -0,0 +1,100 @@
+import {Injectable, OnInit} from '@angular/core';
+import {UserService} from '../userservice';
+import {ApiService} from '../api';
+
+@Injectable()
+export class MessageService {
+  public chat: any = { message: '', newsMessage: ''};
+  public chatMessages: any[] = [];
+  public newsBlock: any[] = [];
+  public urlParams = '';
+  public urlParamsNews = '';
+  private authorId = 0;
+  //
+  constructor(
+    public api: ApiService,
+    public userservice: UserService
+  ) { }
+  public getUserData() {
+    // Retrieve data by key from local storage
+    const userString: string = localStorage.getItem('currentUser');
+    // Return user object if data in local storage exist, or empty object if no user data available
+    return typeof userString === 'string' ? JSON.parse(userString) : {};
+  }
+  public sendMessage(text: string) {
+    const author1: any = { name: 'me', authorId: 1 };
+    const author2: any = { name: 'Some idiot', authorId: 2 };
+    const appUser: any = this.getUserData();
+    const message: any = {
+      text,
+      author: !(this.authorId % 2 && this.authorId !== 0) ? author1.name : author2.name,
+      timestamp: new Date().toISOString()
+    };
+    this.chatMessages.push(message);
+    console.log('message.text', message.text);
+    this.api.post(`/message`, {text: message.text, userId: appUser.id}).subscribe(
+      (resp: any) => {
+        console.log('resp', resp);
+      }
```

Comment:
Add also catching errors, not only success response. Second callback function will give you "error" object that you can use to show some error message to user, like "Error. Message not saved!"